--- /home/atiq/.local/sdkman/bin/sdkman-init.sh.bak	2025-10-04 10:18:45.148851805 -0700
+++ /home/atiq/.local/sdkman/bin/sdkman-init.sh	2025-10-04 10:23:08.878191193 -0700
@@ -113,14 +113,16 @@
 
 export SDKMAN_CANDIDATES_DIR="${SDKMAN_DIR}/candidates"
 
-for candidate_name in "${SDKMAN_CANDIDATES[@]}"; do
-	candidate_dir="${SDKMAN_CANDIDATES_DIR}/${candidate_name}/current"
-	if [[ -h "$candidate_dir" || -d "${candidate_dir}" ]]; then
-		__sdkman_export_candidate_home "$candidate_name" "$candidate_dir"
-		__sdkman_prepend_candidate_to_path "$candidate_dir"
-	fi
+# Iterate in reverse to maintain correct PATH order when appending
+for ((i=${#SDKMAN_CANDIDATES[@]}-1; i>=0; i--)); do
+  candidate_name="${SDKMAN_CANDIDATES[$i]}"
+  candidate_dir="${SDKMAN_CANDIDATES_DIR}/${candidate_name}/current"
+  if [[ -h "$candidate_dir" || -d "${candidate_dir}" ]]; then
+    __sdkman_export_candidate_home "$candidate_name" "$candidate_dir"
+    __sdkman_prepend_candidate_to_path "$candidate_dir"
+  fi
 done
-unset candidate_name candidate_dir
+unset candidate_name candidate_dir i
 export PATH
 
 # source completion scripts
